WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -786,6 +791,23 @@ private void retryUpload(PostModel post, boolean processWithAztec) {
         }
     }
 
+    private ArrayList<MediaModel> getAllFailedMediaForPost(PostModel postModel) {
+        Set<MediaModel> failedMedia = mUploadStore.getFailedMediaForPost(postModel);
+        return filterOutRecentlyDeletedMedia(failedMedia);
+    }
+
+    private ArrayList<MediaModel> filterOutRecentlyDeletedMedia(Set<MediaModel> failedMedia) {
+        ArrayList<MediaModel> mediaToRetry = new ArrayList<>(failedMedia);
```

Comment:
👍  I've refactored it in https://github.com/wordpress-mobile/WordPress-Android/pull/10119/commits/f53a74141cd78bb15324bf2a6ec4aa56cd855ae4 . I used "List" instead of "Set" as the other parts of the code expect a List.